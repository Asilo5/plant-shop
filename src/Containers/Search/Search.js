import React, { Component } from 'react';
import './Search.scss';

class Search extends Component {
    constructor() {
        super();
        this.state = {
            plant : ''
        }
    }

    handleChange = (e) => {
       this.setState({ [e.target.name] : e.target.value })
    }

    render() {
        return (
            <form>
                <button> <span role='img' aria-label='search button'>🔍</span></button>
                <input placeholder='Search' 
                        type='text' 
                        name='plant' 
                        value={this.state.plant}
                        onChange = {(e) => this.handleChange(e)} />
            </form>
        )
    }
}

export default Search;