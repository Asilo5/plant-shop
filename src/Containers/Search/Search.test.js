import React from 'react';
import { shallow } from 'enzyme';
import Search from './Search';

describe('Search Component', () => {
  
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<Search />)
    })
  
    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    })
    
})