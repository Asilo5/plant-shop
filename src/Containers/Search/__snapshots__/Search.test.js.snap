// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component should match snapshot 1`] = `
<form>
  <button>
     
    <FaSearch />
     
  </button>
  <input
    name="plant"
    onChange={[Function]}
    placeholder="Search"
    type="text"
    value=""
  />
</form>
`;
